Index: cfengine3/masterfiles/promises.cf
===================================================================
--- cfengine3.orig/masterfiles/promises.cf	2012-05-31 11:22:49.859532774 -0500
+++ cfengine3/masterfiles/promises.cf	2012-05-31 11:23:13.940008995 -0500
@@ -65,7 +65,7 @@
 
   # End change #
 
-  "dir_masterfiles" string => translatepath("$(sys.workdir)/masterfiles"),
+  "dir_masterfiles" string => translatepath("/var/cfengine/masterfiles"),
                    comment => "Define masterfiles path",
                     handle => "common_def_vars_dir_masterfiles";
 
Index: cfengine3/masterfiles/failsafe.cf
===================================================================
--- cfengine3.orig/masterfiles/failsafe.cf	2012-05-31 11:23:50.892739641 -0500
+++ cfengine3/masterfiles/failsafe.cf	2012-05-31 11:24:14.649210164 -0500
@@ -125,39 +125,6 @@
             action => u_immediate,
            classes => u_if_repaired("update_report");
 
-  !windows::
-
-   "$(sys.workdir)/bin"
-           comment => "Make sure cfengine binaries have right file permissions",
-            handle => "update_files_sys_workdir_bin",
-             perms => u_m("755"),
-      depth_search => u_recurse_basedir("inf"),
-            action => u_immediate;
-
-   "$(sys.workdir)/lib"
-           comment => "Make sure cfengine libraries have right file permissions",
-            handle => "update_files_sys_workdir_lib",
-             perms => u_m("644"),
-      depth_search => u_recurse_basedir("inf"),
-            action => u_immediate;
-
-   "$(sys.workdir)/lib"
-           comment => "Make sure cfengine libraries have right file permissions for only HP-UX",
-            handle => "update_files_sys_workdir_lib_hpux",
-             perms => u_m("755"),
-      depth_search => u_recurse_basedir("inf"),
-            action => u_immediate,
-        ifvarclass => "hpux";
-
-   "/usr/local/sbin"
-           comment => "Ensure cfengine binaries were copied to /usr/local/sbin",
-            handle => "update_files_usr_local_sbin",
-             perms => u_m("755"),
-         copy_from => u_cp_nobck("$(sys.workdir)/bin"),
-       file_select => u_cf3_files,
-      depth_search => u_recurse("1"),
-            action => u_immediate;
-
   am_policy_hub::
 
    "$(master_location)/."
